src/compile/axis/component.ts

```diff
@@ -1,10 +1,10 @@
+import {Axis as VgAxis} from 'vega';
 import {Axis, AxisPart} from '../../axis';
 import {FieldDefBase} from '../../fielddef';
 import {duplicate, Omit} from '../../util';
-import {VgAxis} from '../../vega.schema';
 import {Split} from '../split';
 
-function isFalseOrNull(v: boolean | null) {
+function isFalseOrNull(v: any) {
```

Comment:
Why do we need to change the type here?

Reply:
This can be a BooleanValue now because axis config is a VgAxis (which has potentially signals).

Reply:
Hmm, shouldn't BooleanValue be not allowed for Vega-Lite?

Reply:
It is allowed to be generated in a component, though.

Reply:
An axis component is a generic that is typed with VgAxis. An axis component could use signals in it's definition (and it does in fact for `tickCount`)